libguard/guard_file.cpp

```diff
@@ -22,22 +24,28 @@ GuardFile::GuardFile(const fs::path& file) : guardFile(file)
         {
             guard_log(GUARD_ERROR,
                       "Failed to open the GUARD file during initailization");
-            throw std::runtime_error("Failed to get the GUARD file.");
+            throw GuardFileOpenFailed("Failed to get the GUARD file.");
         }
         file.seekg(0, file.end);
         if (file.fail())
         {
-            throw std::runtime_error(
+            throw InvalidEntityPath(
                 "Failed to move to the last position in the file");
         }
         fileSize = file.tellg();
     }
-    catch (const std::exception& ex)
+    catch (const GuardFileOpenFailed& ex)
```

Comment:
- Already you are throwing exceptions in the above so why catching and throwing the same exception? 
- Also too many traces are possible for same exception

Reply:
Done.